core-java-modules/core-java-8-2/src/test/java/com/baeldung/javaClass/ClassForNameAndInstanceUnitTest.java

```diff
@@ -0,0 +1,19 @@
+package com.baeldung.javaClass;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class ClassForNameAndInstanceUnitTest {
+    ClassForNameAndInstance tester= new ClassForNameAndInstance();
+    @Test
+    public void givenString_whenValidClass_thenReturnClass() throws ClassNotFoundException {
+        Class c= tester.createClassForName("java.lang.String");
+        assertEquals(c.getName(),"java.lang.String");
```

Comment:
This assertion is backwards. We should put the expected value at the beginning.

It might be better if we did `assertEquals(String.class, c)` and maybe even call the variable `stringClass`